src/Dlp/V2beta1/Gapic/DlpServiceGapicClient.php

```diff
@@ -141,55 +184,48 @@ class DlpServiceGapicClient
      * @return string The formatted result resource.
      * @experimental
      */
-    public static function formatResultName($result)
+    public static function resultName($result)
     {
         return self::getResultNameTemplate()->render([
             'result' => $result,
         ]);
     }
 
     /**
-     * Parses the result from the given fully-qualified path which
-     * represents a result resource.
+     * Parses a formatted name string and returns an associative array of the components in the name.
+     * The following name formats are supported:
+     * Template: Pattern
+     * - result: inspect/results/{result}.
+     *
+     * The optional $template argument can be supplied to specify a particular pattern, and must
+     * match one of the templates listed above. If no $template argument is provided, or if the
+     * $template argument does not match one of the templates listed, then parseName will check
+     * each of the supported templates, and return the first match.
+     *
+     * @param string $formattedName The formatted name string
+     * @param string $template      Optional name of template to match
      *
-     * @param string $resultName The fully-qualified result resource.
+     * @return array An associative array from name component IDs to component values.
      *
-     * @return string The extracted result value.
+     * @throws ValidationException If $formattedName could not be matched.
      * @experimental
      */
-    public static function parseResultFromResultName($resultName)
+    public static function parseName($formattedName, $template = null)
     {
-        return self::getResultNameTemplate()->match($resultName)['result'];
-    }
+        $templateMap = self::getPathTemplateMap();
 
-    private static function getResultNameTemplate()
-    {
-        if (self::$resultNameTemplate == null) {
-            self::$resultNameTemplate = new PathTemplate('inspect/results/{result}');
+        if (isset($templateMap[$template])) {
```

Comment:
I think an exception here if `$templateMap[$template]` is not set is preferable. As it stands, one could continue under the assumption they've optimized to avoid the loop when in reality all they've done is provided an invalid value.

```php
if ($template) {
    if (!isset($templateMap[$template])) {
        throw new \Exception(sprintf('Template name %s does not exist!', $template));
    }

    return $templateMap[$template]->match($formattedName);
}
```

note: previous copy of this comment was in a weird spot. sorry for the dupe.

Reply:
Done